source/includes/connection-snippets/x509-snippets/c-connection.c

```diff
@@ -0,0 +1,37 @@
+// begin x509 connection
+#include <mongoc/mongoc.h>
+
+mongoc_client_t * client;
+mongoc_database_t * database;
+mongoc_collection_t * collection;
+bson_t * command, reply, * insert;
+bson_error_t error;
+char * str;
+bool retval;
+
+mongoc_init();
+
+const mongoc_ssl_opt_t * ssl_default = mongoc_ssl_opt_get_default();
+mongoc_ssl_opt_t ssl_opts = {
+  0
+};
+
+memcpy( & ssl_opts, ssl_default, sizeof ssl_opts);
+
+ssl_opts.pem_file = "/etc/certs/mongodb/client.pem";
+ssl_opts.ca_file = "/etc/certs/mongodb/ca.pem";
+ssl_opts.weak_cert_validation = false;
+ssl_opts.allow_invalid_hostname = false;
+
+char subject[] = "mongodb://<your_subject>"; // mongodb://CN%3DChris%2COU%3DTestClientCertificateOrgUnit%2CO%3DEducationClientCertificate%2CL%3DTestClientCertificateLocality%2CST%3DTestClientCertificateState%2CC%3DUS
```

Comment:
Feel free to remove the comment from this line in this repo.

Reply:
Do you think people would be confused about what "<your_subject>" means without the comment? @ccho-mongodb

Reply:
You could use:
"x509 client certificate subject" and also comment that it should be urlencoded and provide a link to a sample. If no samples exist, maybe you could create a comment line that contains a sample (the current values are not great -- I think openssl provides some default values when you create a certificate which you could use instead)

Reply:
The username is also unnecessary since it's included in the certificate already. If you omit it when connecting to mongodb it'll just assign you a username matching the subject. Why introduce another point of failure for users?

Reply:
I've altered this snippet to more closely match the others. Let me know what you think @nathan-contino-mongo

Reply:
Removed chris's x509 cert subject with the comment `// x509 client certificate subject`